core/config/project_settings.cpp

```diff
@@ -501,7 +501,84 @@ Error ProjectSettings::_setup(const String &p_path, const String &p_main_pack, b
+	if (!p_main_pack.is_empty()) {
+		bool ok = _load_resource_pack(p_main_pack, true, 0, true);
+		ERR_FAIL_COND_V_MSG(!ok, ERR_CANT_OPEN, "Cannot open resource pack '" + p_main_pack + "'.");
+
+		Error err = _load_settings_text_or_binary("res://project.godot", "res://project.binary");
+		if (err == OK && !p_ignore_override) {
+			// Load override from location of the main pack
+			// Optional, we don't mind if it fails
+			_load_settings_text(p_main_pack.get_base_dir().path_join("override.cfg"));
+		}
+		return err;
+	}
+
+	String exec_path = OS::get_singleton()->get_executable_path();
+
+	if (!exec_path.is_empty()) {
+		// We do several tests sequentially until one succeeds to find a PCK,
+		// and if so, we attempt loading it at the end.
+
+		// Attempt with PCK bundled into executable.
+		bool found = _load_resource_pack(exec_path, true, 0, true);
+
+		// Attempt with exec_name.pck.
+		// (This is the usual case when distributing a Godot game.)
+		String exec_dir = exec_path.get_base_dir();
+		String exec_filename = exec_path.get_file();
+		String exec_basename = exec_filename.get_basename();
+
+		// Based on the OS, it can be the exec path + '.pck' (Linux w/o extension, macOS in .app bundle)
+		// or the exec path's basename + '.pck' (Windows).
+		// We need to test both possibilities as extensions for Linux binaries are optional
+		// (so both 'mygame.bin' and 'mygame' should be able to find 'mygame.pck').
+
+#ifdef MACOS_ENABLED
+		if (!found) {
+			// Attempt to load PCK from macOS .app bundle resources.
+			found = _load_resource_pack(OS::get_singleton()->get_bundle_resource_dir().path_join(exec_basename + ".pck"), true, 0, true) || _load_resource_pack(OS::get_singleton()->get_bundle_resource_dir().path_join(exec_filename + ".pck"), true, 0, true);
+		}
+#endif
+
+		if (!found) {
+			// Try to load data pack at the location of the executable.
+			// As mentioned above, we have two potential names to attempt.
+			found = _load_resource_pack(exec_dir.path_join(exec_basename + ".pck"), true, 0, true) || _load_resource_pack(exec_dir.path_join(exec_filename + ".pck"), true, 0, true);
+		}
+
+		if (!found) {
+			// If we couldn't find them next to the executable, we attempt
+			// the current working directory. Same story, two tests.
+			found = _load_resource_pack(exec_basename + ".pck", true, 0, true) || _load_resource_pack(exec_filename + ".pck", true, 0, true);
+		}
+
+		// If we opened our package, try and load our project.
+		if (found) {
+			Error err = _load_settings_text_or_binary("res://project.godot", "res://project.binary");
+			return err;
+		}
+	}
+
+#ifdef ANDROID_ENABLED
+	// Try to use the filesystem for files, according to OS.
+	// (Only Android -when reading from APK- use this.)
+
+	if (!OS::get_singleton()->get_resource_dir().is_empty()) {
+		Error err = _load_settings_text_or_binary("res://project.godot", "res://project.binary");
+		return err;
+	}
+#endif
+
+	return ERR_CANT_OPEN;
+
+#else
```

Comment:
This  seems to be a repeat of the logic below, the only difference being the `p_require_encryption` parameter value.
Can you wrap the logic in an helper method instead; this would improve readability and ensure that the two code path don't accidentally diverge.